tests/unittests/test_main.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+
+from azure_functions_worker.main import create_args_parser
+
+
+class TestMain(unittest.TestCase):
+
```

Comment:
Have to tried mocking the sys.argv? 
Something like this.
```
@patch.object(sys, 'argv', ['xxx', '--host', 'localhost',])
    def test_parse_args(self):
        args = parse_args()
        self.assertEqual(args.host, 'localhost')
```

In this way you wont have to change the code in main.

Reply:
updated all tests.